File: Codigo_Fonte_pascal/lista2/lista2_7.pas
program Fibonacci;

var
   a, b, fib, cont : integer;

begin
     a := 0;
     b := 1;

     //writeln('Estrutura de repetição while...do');
     cont := 1;
     write(a,', ',b);
     while cont <= 8 do
     begin
          //writeln(cont);
          fib := a + b;
          write(', ', fib);
          a := b;
          b := fib;
          cont := cont + 1;
     end;
     writeln;


     //writeln('Estrutura de repetição repeat...until');
     a := 0;
     b := 1;
     cont := 1;
     write(a,', ',b);
     repeat
           //writeln(cont);
           fib := a + b;
           write(', ', fib);
           a := b;
           b := fib;
           cont := cont + 1;
     until cont > 8;

     readln;

end.

File: Codigo_Fonte_c/Lista 2/lista2_7.c
//Exibir os dez primeiros números da série de Fibonacci.
#include <stdio.h>

int main() {
	int a, b, fib;
	a = 0;
	b = 1;

// estrutura de repetição while
	int cont = 1;
	printf("%d, %d", a, b);
	while (cont <= 8){
		fib = a + b;
		printf(", %d", fib);
		a = b;
		b = fib;
		cont++ ;
	}
	printf("\n");
	
	
// estrutura de repetição do while
	a = 0;
	b = 1;
	cont = 1;
	printf("%d, %d", a, b);
	do
	{
		fib = a + b;
		printf(", %d", fib);
		a = b;
		b = fib;
		cont++ ;
	}while(cont <= 8);

	printf("\n");
	return 0;
}
